src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestSystemBuildingDataStore.java

```diff
@@ -0,0 +1,212 @@
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.NBTUtils;
+import com.minecolonies.api.util.constant.TypeConstants;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+
+public class StandardRequestSystemBuildingDataStore implements IRequestSystemBuildingDataStore
+{
+
+    private IToken<?>                                id;
+    private final Map<TypeToken<?>, Collection<IToken<?>>> openRequestsByRequestableType;
+    private final Map<Integer, Collection<IToken<?>>>      openRequestsByCitizen;
+    private final Map<Integer, Collection<IToken<?>>>      completedRequestsByCitizen;
+    private final Map<IToken<?>, Integer>                  citizenByOpenRequest;
+
+    public StandardRequestSystemBuildingDataStore(
+      final IToken<?> id,
+      final Map<TypeToken<?>, Collection<IToken<?>>> openRequestsByRequestableType,
+      final Map<Integer, Collection<IToken<?>>> openRequestsByCitizen,
+      final Map<Integer, Collection<IToken<?>>> completedRequestsByCitizen,
+      final Map<IToken<?>, Integer> citizenByOpenRequest) {
+        this.id = id;
+        this.openRequestsByRequestableType = openRequestsByRequestableType;
+        this.openRequestsByCitizen = openRequestsByCitizen;
+        this.completedRequestsByCitizen = completedRequestsByCitizen;
+        this.citizenByOpenRequest = citizenByOpenRequest;
+    }
+
+    public StandardRequestSystemBuildingDataStore()
+    {
+        this(StandardFactoryController.getInstance().getNewInstance(TypeConstants.ITOKEN),
+          new HashMap<>(),
+          new HashMap<>(),
+          new HashMap<>(),
+          new HashMap<>());
+    }
+
+    @Override
+    public Map<TypeToken<?>, Collection<IToken<?>>> getOpenRequestsByRequestableType()
+    {
+        return openRequestsByRequestableType;
+    }
+
+    @Override
+    public Map<Integer, Collection<IToken<?>>> getOpenRequestsByCitizen()
+    {
+        return openRequestsByCitizen;
+    }
+
+    @Override
+    public Map<Integer, Collection<IToken<?>>> getCompletedRequestsByCitizen()
+    {
+        return completedRequestsByCitizen;
+    }
+
+    @Override
+    public Map<IToken<?>, Integer> getCitizensByRequest()
+    {
+        return citizenByOpenRequest;
+    }
+
+    @Override
+    public IToken<?> getId()
+    {
+        return id;
+    }
+
+    @Override
+    public void setId(final IToken<?> id)
+    {
+        this.id = id;
+    }
+
+    public static class Factory implements IFactory<FactoryVoidInput, StandardRequestSystemBuildingDataStore>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 101 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)